### PR TITLE
pio: init at 0.4.0

### PR DESCRIPTION
**Pull request by Liassica**
## Description of changes

[Pio](https://github.com/siiptuo/pio) is a command line tool for optimizing PNG, WebP, and JPEG images.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/pi/pio/package.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, rustPlatform, fetchFromGitHub, stdenv }:
+rustPlatform.buildRustPackage rec {
```
Comment (AndersonTorres) [inline pkgs/by-name/pi/pio/package.nix:2]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, stdenv }:

rustPlatform.buildRustPackage rec {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
```json
{
    "blacklisted": [],
    "broken": [],
    "built": [
        "pio"
    ],
    "extra-nixpkgs-config": null,
    "failed": [],
    "non-existent": [],
    "pr": 280590,
    "system": "x86_64-linux",
    "tests": []
}
```

**Review by Janik-Haag — COMMENTED** [review]
please squash the `pio:` commits into `pio: init at 0.4.0` since nixpkgs tries to keep a atomic commit history, otherwise this seems good to go.

File: pkgs/by-name/pi/pio/package.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+rustPlatform.buildRustPackage rec {
+  pname = "pio";
+  version = "0.4.0";
+
+  src = fetchFromGitHub {
+    owner = "siiptuo";
+    repo = "pio";
+    rev = version;
+    hash = "sha256:1mmf4hyjjlm34ciabnmkvsc4gbcxywfcldxjhi9ksfaldpw8c7l9";
```
Comment (h7x4) [inline pkgs/by-name/pi/pio/package.nix:11]:
```suggestion
    hash = "sha256-1mmf4hyjjlm34ciabnmkvsc4gbcxywfcldxjhi9ksfaldpw8c7l9";
```

File: pkgs/by-name/pi/pio/package.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  cargoHash = "sha256-jVOpk+Z3yEEoDexvxT9I0aVHJKVq47y8km/9ltoqrDA=";
+
+  meta = with lib; {
+    description = "Utility to compress image files while maintaining quality";
+    homepage = "https://github.com/siiptuo/pio";
+    changelog = "https://github.com/siiptuo/pio/blob/${version}/CHANGELOG.md";
+    license = licenses.agpl3Plus;
+    maintainers = with maintainers; [ liassica ];
```
Comment (h7x4) [inline pkgs/by-name/pi/pio/package.nix:21]:
```suggestion
    maintainers = with maintainers; [ liassica ];
    mainProgram = "pio";
```

**Review by Janik-Haag — APPROVED** [review]

**Comment by h7x4** [comment]
Would you mind signing at least the first commit? We can't [verify your keys](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md#individual-maintainer-list) without it.

**Comment by Liassica** [comment]
The commits are signed, it's just GitHub won't let me add my committing email to my account so it shows as "unverified". My key fingerprint is on my profile (and in the maintainers list from that commit), so feel free to manually verify. As per the documentation you linked:
> Note: GitHub's "Verified" label does not display the user's full key fingerprint, and should not be used for validating the key matches.

**Review by h7x4 — APPROVED** [review]
I've checked the `githubId`, verified the gpg key, and tested the program. Looks good!

File: pkgs/by-name/pi/pio/package.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{ lib, rustPlatform, fetchFromGitHub, stdenv }:
```
Comment (h7x4) [inline pkgs/by-name/pi/pio/package.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub }:
```

**Comment by Janik-Haag** [comment]
one last thing, I just saw that the build fails on darwin, so you should either fix that or mark it as broken.

**Comment by Liassica** [comment]
Added the "Accelerate" framework as a Darwin build input as the logs seemed to indicate was needed, but I don't have a Darwin machine to test on so my only indication if it works is if the build test passes.

**Comment by Liassica** [comment]
Seems to have worked.

**Pull request merged**